pkg/controller/job/job_controller.go

```diff
@@ -1059,11 +1072,13 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 				needsFlush = true
 				uncountedStatus.Succeeded = append(uncountedStatus.Succeeded, pod.UID)
 			}
-		} else if considerPodFailed || jobCtx.finishedCondition != nil {
+		} else if considerPodFailed || (jobCtx.finishedCondition != nil &&
+			!(feature.DefaultFeatureGate.Enabled(features.JobSuccessPolicy) && jobCtx.finishedCondition.Type == batch.JobSuccessCriteriaMet)) {
```

Comment:
maybe if we had `jobCtx.successCondition` this could be expressed easier?

Reply:
Yes, that's right.

Reply:
Done.